src/main/java/org/jboss/set/ValidationCommand.java

```diff
@@ -0,0 +1,147 @@
+            GHPullRequest pullRequest = ghRepo.getPullRequest(pullRequestId);
+            for (GHPullRequestFileDetail changedFile : pullRequest.listFiles()) {
+                if (changedFile.getFilename().equals(".github/" + LotteryConfig.FILE_NAME)) {
+                    Log.infof("Pull Request %d tries to change %s file", pullRequestId, LotteryConfig.FILE_NAME);
+                    GHContent updatedFile = gitHub.getRepository(pullRequest.getHead().getRepository().getFullName())
+                            .getFileContent(".github/"
+                                    + LotteryConfig.FILE_NAME, pullRequest.getHead().getSha());
+                    String updatedFileContent = new String(updatedFile.read().readAllBytes());
+                    LotteryConfig lotteryConfig = lotteryConfigProducer.getLotteryConfigFromString(updatedFileContent);
+                    if (validateConfig(lotteryConfig)) {
+                        Log.infof("Pull Request %d keeps the config file valid", pullRequestId);
+                    } else {
+                        Log.infof("Pull Request %d doesn't keep the config file valid", pullRequestId);
+                        // This will cause unsuccessful exit code, thus failing the CI/CD pipeline action
+                        throw new RuntimeException("The file was not validated successfully");
+                    }
+                }
+            }
+        } catch (IOException | URISyntaxException e) {
+            errorAndPrint(e, "Unable to open anonymous connection to GitHub or parse URL of the repository");
+            throw new RuntimeException(e);
+        }
+    }
+
+    private boolean validateConfig(LotteryConfig lotteryConfig) {
+        Map<String, Collection<String>> projectComponentsMap = new HashMap<>();
+        Set<String> participants = new HashSet<>();
+        boolean valid = true;
+
+        for (LotteryConfig.Participant participant : lotteryConfig.participants()) {
+
+            if (participant.maxIssues() != null && participant.maxIssues() <= 0) {
+                errorAndPrint("Participant %s has defined non-positive number of max issues".formatted(participant.email()));
+                valid = false;
+            }
+
+            for (LotteryConfig.Participant.Project project : participant.projects()) {
+                if (participants.contains(participant.email())) {
+                    errorAndPrint("Two identical participants detected %s".formatted(participant.email()));
+                }
+                participants.add(participant.email());
+
+                if (project.participation().maxIssues() <= 0) {
+                    errorAndPrint("Participant %s has defined non-positive number of max issues for project %s"
+                            .formatted(participant.email(), project.project()));
+                    valid = false;
+                }
+
+                // we assume a valid project has some components
```

Comment:
It would be better to assume all components if none is defined.

Reply:
This is when we call JIRA backend, to retrieve info about a project. Now, I will catch the exception of non-existing project, and set it's components for  empty list.